### DISCUSSION TITLE
Is there a way to add a new file to all of my projects in my Git Repository at once?

### DISCUSSION THREAD

**Discussion by deepu558** [community/community#21895] [New to GitHub]
<p>Hi, I am trying to add a config file to all my projects in my Git repository.  I can do it one by one but I had more than 140 projects in my Git repo, which takes a lot of time to get that done, so I was trying to find a shorter way to add this file to all of my projects at once, is it possible? by writing any script?</p>

**Comment by lee-dohm** [comment]
<p>Do you mean that you have a bunch of subdirectories under <a href="https://github.com/deepu558/some-repo-name" rel="nofollow noopener">https://github.com/deepu558/some-repo-name</a> that you want to add the same file to? Or do you mean that you have a bunch of repositories under <a href="https://github.com/deepu558" rel="nofollow noopener">https://github.com/deepu558</a> that you want to add the same file to?</p>

**Comment by deepu558** [comment]
<p>I mean that I have a bunch of repositories under <a href="https://github.com/deepu558" rel="nofollow noopener">https://github.com/deepu558</a> that I want to add the same file to.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>There isn’t a built-in way to do this on GitHub. When I’ve needed to do something like this, I’ve built a script to loop through all the repositories listed in a text file and perform the following with each:</p>
<ol>
<li>Clone the repository to my local machine if it doesn’t exist already</li>
<li>Perform a <code>git fetch</code>
</li>
<li>Create a new branch based on the default branch</li>
<li>Copy the file from a given location into the repository</li>
<li>Commit the change</li>
<li>Push the new branch to the remote repository</li>
<li>Launch the link to create a new PR based on the new branch in my default browser</li>
<li>Wait for me to press a key before moving to the next one</li>
</ol>
<p>I posted <a href="https://gist.github.com/lee-dohm/b882305665229f51d16a542953420255" rel="nofollow noopener">a Ruby script that does a lot of the Git interactions</a> to help get you started.</p>
